tests/codegen/targets/cpp_test.py

```diff
@@ -0,0 +1,196 @@
+import unittest
+from functools import reduce
+from operator import mul
+
+import dace
+from dace import SDFG, Memlet
+from dace.codegen.targets import cpp
+from dace.subsets import Range
+from dace.transformation.dataflow import RedundantArray
+
+
+class ReshapeStrides(unittest.TestCase):
+    def test_multidim_array_all_dims_unit(self):
+        r = Range([(0, 0, 1), (0, 0, 1)])
+
+        # To smaller-sized shape
+        target_dims = [1]
+        self.assertEqual(reduce(mul, r.size_exact()), reduce(mul, target_dims))
```

Comment:
you should use the pytest `assert reduce(...) == reduce(...)`

Reply:
Done.

Just curious: Is there a downside of `unittest` or is it just to be consistent with the rest of the code?
As far as I can tell, they are compatible with `pytest`, offers better organisation of code, and the better error messages. Then again, I'm not very familiar with python testing frameworks, so perhaps I've missed some disadvantages.